tests/foreman/ui/test_provisioningtemplate.py

```diff
@@ -51,31 +70,24 @@ def test_positive_clone(session):
 
     :CaseLevel: Integration
     """
-    name = gen_string('alpha')
     clone_name = gen_string('alpha')
-    content = gen_string('alpha')
-    os_list = [
-        entities.OperatingSystem().create().title for _ in range(2)
-    ]
     with session:
-        session.provisioningtemplate.create({
-            'template.name': name,
-            'template.template_editor.editor': content,
-            'type.template_type': 'Provisioning template'
-        })
-        assert session.provisioningtemplate.search(name)[0]['Name'] == name
         session.provisioningtemplate.clone(
-            name,
+            clone_setup['pt'].name,
             {
                 'template.name': clone_name,
-                'association.applicable_os.assigned': os_list
+                'association.applicable_os.assigned': clone_setup['os']
             }
         )
-        assert session.provisioningtemplate.search(
-            clone_name)[0]['Name'] == clone_name
-        template = session.provisioningtemplate.read(clone_name)
-        assert set(os_list) == set(
-            template['association']['applicable_os']['assigned'])
+        pt = entities.ProvisioningTemplate().search(
+            query={'search': 'name=={0}'.format(clone_name)}
+        )
+        assigned_oses = [os.read() for os in pt[0].read().operatingsystem]
+        assert len(pt) > 0, \
```

Comment:
Why `len`? In other places, you assume `search()` call that returns something will be truthy, while these that do not return anything will be falsey.

It doesn't really matter, but I think we should be consistent.